File: app/models/Page.scala
package models

sealed abstract class Page(
  val key: String,
  val path: String,
  val title: String
)

object Page {
  case object Index extends Page("index", "/", "トップページ")
  case object Program extends Page("program", "/program", "プログラミング")
  case object Anime extends Page("anime", "/anime", "アニメ")
  case object LightNovel extends Page("light_novel", "/light_novel", "ライトノベル")
  case object Link extends Page("link", "/link", "リンク")
  case object SiteInfo extends Page("site_info", "/site_info", "サイト情報")

  val values = Array(Index, Program, Anime, LightNovel, Link, SiteInfo)

  def of(key: String) = this.values.find(_.key == key)
}

File: app/src/main/java/moe/pine/profile/models/Page.java
package moe.pine.profile.models;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum Page {
    INDEX("index", "/", "トップページ", "Top Page"),
    PROGRAM("program", "/program", "プログラミング", "Programming"),
    ANIME("anime", "/anime", "アニメ", "Anime"),
    LIGHT_NOVEL("light_novel", "/light_novel", "ライトノベル", "Light novel"),
    MOUNTAIN("mountain", "/mountain", "登山", "Mountain"),
    LINK("link", "/link", "リンク", "Link"),
    SITE_INFO("site_info", "/site_info", "サイト情報", "Site information"),
    ;

    @Getter
    private final String template;

    @Getter
    private final String path;

    @Getter
    private final String title;

    @Getter
    private final String titleEn;
}
